### DISCUSSION TITLE
Cash

### DISCUSSION THREAD

**Discussion by elexx** [apauley/hledger-flow#117] [Q&A] [answered]
Hello.
I'm trying to wrap my head around a cash/non-digital account. Unfortunately my purse does not have a CSV export ;-) , so I have to write my cash journal by hand. I tried a few things: 
* Writing my own `import/elexx/_manual_/2023/pre-import.journal`, however it feels a bit off to write my `cash.journal` in a file named `pre-import.journal`. Also I'm not sure how a would add a second non-digital account with this approach.
* Writing my cash journal as csv files `import/elexx/cash/cash/2023/[01|02|...12]-cash.csv` and importing them as I would import any other CSV exported data. However this feels even more off. Because now I'm writing CSVs by hand with `.rules` to parse them.

Somehow I feel I missed something in the docs and there is an obvious, easy way I should follow. I'd appreciate some input on how to approach this problem. Thanks!

**Comment by elexx** [comment, marked as answer]
I write a CSV for my cash manually (import/elexx/manual/cash), the rules file looks like
```
separator    ,
date-format  %Y-%m-%d
decimal-mark .
currency     EUR
fields       date, description, amount1, account2
account1     assets:Cash
```

I also stumbled into another problem, when lending/borrowing money. For example if a friend of mine orders food for all of us and I transfer money from my bank account to his. In this case I have a rule in my bank account rules that sets the second account for this transaction to a "...:transfer:my-friends-name". And to finally move the money from this temporary account to the "expenses:food" account a write another CSV manually (import/elexx/manual/transfer) with this ruleset:
```
separator    ,
date-format  %Y-%m-%d
decimal-mark .
currency     EUR
fields       date, description, amount1, account1, account2
```
If anyone is interested in a full example/template of this setup, let me know and I'll set up a repo with it.

**Comment by sbibauw** [comment]
Some other options for cash accounts:
- If most of your cash expenses are in the same category (e.g., `expenses:food`), just assume that every time you take cash at the ATM, it is immediately spent for food (rule in your bank account: `if <ATM>; account2 expenses:food`).
- If your cash expenses are many small amounts, you could still do that, but with a dedicated category (e.g., `expenses:cash`): each ATM withdrawal moves money from `assets:bank:x` to `expenses:cash`. If, occasionally, you spend a lot of cash on something that you want to distinguish, let's say because you bought a car in cash, then you can write a manual transaction moving the money from `expenses:cash` to `expenses:car`:
```
2024-03-01 Car
  expenses:cash   -10000.00
  expenses:car     10000.00
```    

Finally, to manually add some transactions, you can indeed create a CSV with importation rules, but the most direct workflow would be to either write the hledger `.journal` file directly (not much longer than a CSV), or, better, to use an interactive tool such as [hledger-iadd](https://github.com/hpdeifel/hledger-iadd) .

**Reply by elexx** [reply]
I really like the idea of using already existing tools like iadd, however I'm not sure into which journal file I should put the transactions. What would you suggest?

**Reply by sbibauw** [reply]
Apart from my normally imported folders and journals (e.g., `me/bank/account/1-in/2024/x.csv`), I have a `_manual_/2024/post-import.journal` where I put all manually added transactions. These transactions can be cash expenses or manual corrections for some auto assignments of bank imports, or anything you like to keep track of manually.
hledger-flow ignores the `_manual_` folder for import workflow.
